src/external_data_ifaces_impl.cpp

```diff
@@ -109,6 +111,42 @@ sdbusplus::async::task<> ExternalDataIFacesImpl::fetchBMCPosition()
     co_return;
 }
 
+sdbusplus::async::task<>
+    ExternalDataIFacesImpl::createErrorLog(const std::string& errMsg,
+                                           const ErrorLevel& errSeverity,
+                                           const json& calloutsDetails)
+{
+    try
+    {
+        error_log::FFDCFileInfoSet ffdcFileInfoSet;
+        error_log::FFDCFile::createCalloutFFDC(calloutsDetails,
```

Comment:
It could be simplified to avoid a function in the FFDCClass (its creating object for the same class).

```
if (!calloutData.is_null())
{
    FFDCFile file(FFDCFormat::JSON, 0xCA, 0x01, calloutData.dump());
    ffdcFileInfoSet.emplace_back(file.getFormat(), file.getSubType(),
                                 file.getVersion(), file.getFD());
}